parts/k8s/kubernetesagentcustomdata.yml

```diff
@@ -34,14 +34,14 @@ write_files:
       "log-opts":  {
          "max-size": "50m",
          "max-file": "5"
-      }{{if IsNVIDIADevicePluginEnabled}}
+      }{{if IsNSeriesSKU .}}{{if IsNVIDIADevicePluginEnabled}}
```

Comment:
Let's figure out how to simplify these guards.

1. are there scenarios where we *don't* want to install the nvidia device plugin on an N Series VM SKU node?
2. are there scenarios where we would ever want to install the plugin on a non-N Series VM SKU node?

I infer the answer to #2 is "no" based on the way this has been implemented here. For #1, if the answer to that is also "no", then we should combine these two checks: basically, we should get rid of IsNVIDIADevicePluginEnabled, or we should have IsNVIDIADevicePluginEnabled only return true if it's on an N Series node.

Reply:
The answer to #1 is Yes. We basically didn't want to break backwards compatibility and leave clusters < 1.10 the way they are currently working. Once we no longer support k8s 1.9 then the answer to #1 is No.